### PR TITLE
feat(fivetran): app install, sync users & delete user 

### PR DESCRIPTION
**Pull request by athielba**
## Description

1.  Input authorization credentials
2. sync users
3. Schedule sync users
4. delete user

Related [ticket](https://linear.app/elba-integrations/team/FIV/all)
## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/fivetran/src/inngest/functions/users/delete-users.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { db } from '@/database/client';
+import { organisationsTable } from '@/database/schema';
+import { inngest } from '@/inngest/client';
+import { deleteUser } from '@/connectors/users';
+import { env } from '@/env';
+import { decrypt } from '@/common/crypto';
+
+export const deleteSourceUser = inngest.createFunction(
+  {
+    id: 'fivetran-delete-users',
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/functions/users/delete-users.ts:12]:
```suggestion
    id: 'fivetran-delete-user',
```

File: apps/fivetran/src/inngest/functions/users/delete-users.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: env.FIVETRAN_USERS_DELETE_CONCURRENCY,
```
[the remark this answers is not in the document]
Comment (elbalexandre) [reply apps/fivetran/src/inngest/functions/users/delete-users.ts:15]: I think I choose this name for others integration.

File: apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { env } from '@/env';
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts:1]: file should be named `schedule-users-syncs`

File: apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { db } from '@/database/client';
+import { organisationsTable } from '@/database/schema';
+import { inngest } from '../../client';
+
+export const scheduleUsersSynchronize = inngest.createFunction(
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts:6]:
```suggestion
export const scheduleUsersSyncs = inngest.createFunction(
```

File: apps/fivetran/src/inngest/functions/users/sync-users.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      const users = result.validUsers.filter(({ active }) => active).map(formatElbaUser);
+
+      if (result.invalidUsers.length > 0) {
+        logger.warn('Retrieved users contains invalid data', {
+          organisationId,
+          invalidUsers: result.invalidUsers,
+        });
+      }
+
+      await elba.users.update({ users });
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/functions/users/sync-users.ts:89]:
```suggestion
      if (users.length > 0) {
        await elba.users.update({ users });
      }
```

File: apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+          transformOutput: (ctx) => {
+            const {
+              result: { error, ...result },
+              ...context
+            } = ctx;
+
+            // Check if the error is a rate limit error (HTTP Status 429)
+            if (error instanceof FivetranError && error.response?.status === 429) {
+              // This is a simplified approach; adjust logic based on actual rate limit policy details
+              const retryAfter = error.response.headers['Retry-After'] as string; // Retry after the interval if no remaining calls
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts:19]:
```suggestion
              const retryAfter = error.response.headers.get('Retry-After') || 60; // Retry after the interval if no remaining calls
```

File: apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+              return {
+                ...context,
+                result: {
+                  ...result,
+                  error: new RetryAfterError(
+                    `Rate limit exceeded for '${fn.name}'. Retry after ${retryAfter} seconds.`,
+                    retryAfter,
```
Comment (elbalexandre) [inline apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts:27]:
```suggestion
                    `${retryAfter}s`,
```

File: apps/fivetran/src/inngest/functions/users/delete-users.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      .select({
+        apiKey: organisationsTable.apiKey,
+        apiSecret: organisationsTable.apiSecret,
+      })
+      .from(organisationsTable)
+      .where(eq(organisationsTable.id, organisationId));
+
+    if (!organisation) {
+      throw new NonRetriableError(
+        `API key & Secret not found for organisation with ID: ${organisationId} for the user id ${userId}`
```
[the remark this answers is not in the document]
Comment (athielba) [reply apps/fivetran/src/inngest/functions/users/delete-users.ts:33]: Make sense

File: apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts (hunk 3 above)
Comment (athielba) [reply apps/fivetran/src/inngest/functions/users/schedule-users-sync.ts:1]: I am not sure, what it should be in plural ?

**Review by proquig — APPROVED** [review]

**Review by proquig — APPROVED** [review]

**Pull request merged**